akka-docs/src/main/paradox/general/stream/stream-design.md

```diff
@@ -45,6 +45,27 @@ All stream Processors produced by the default materialization of Akka Streams ar
+Akka Streams was designed during the development of Reactive Streams, so they both heavily influenced one another.
+
+It may be sunrising to learn that even within the Reactive Specification the types had initially attempted to hide `Publisher`, `Subscriber` and the other SPI types from users of the API.
+Though since those internal SPI types would end up surfacing to end users of the standard in some cases, it was decided to [remove the API types, and only keep the SPI types][9] which are the `Publisher`, `Subscriber` et al.
+
+With this historical knowledge and context about the purpose of the standard – being an internal detail of interoperable libraries - we can with certainty say that it can't be really said that an direct _inheritance_ relationship with these types can be considered some form of advantage or meaningful differentiator between libraries.
+Rather, it could be seen that APIs which expose those SPI types to end-users are leaking internal implementation details accidentally. 
+
+The `Source`, `Sink` and `Flow` types which are part of Akka Streams, have their direct counter parts in Reactive Streams (being, respectively, `Publisher`, `Subscriber` and `Processor`), and their purpose is to provide the fluent DSL,
+as well as be "factories" for running those streams. In other words, Akka Streams operate on a [lifted representation](https://doc.akka.io/docs/akka/2.5.6/scala/general/stream/stream-design.html) of the computing graph,
+which then is materialized and executed in accordance to Reactive Streams rules. This also allows Akka Streams to perform optimizations like fusing and dispatcher configuration during the materialization step.
+
+Another not obvious gain from hiding the Reactive Streams interfaces comes from the fact that `org.reactivestreams.Subscriber` (et al) have now been included in Java 9+, and thus become part of Java itself, so libraries should migrate to using the `java.util.concurrent.Flow.Subscriber` instead of `org.reactivestreams.Subscriber`.
+Libraries which selected to expose and directly extend the Reactive Streams types will now have a tougher time to adapt the JDK9+ types -- all their classes that extend Subscriber and friends will need to be copied or changed to extend the exact same interface,
+but from a different package. In Akka we simply expose the new type when asked to -- already supporting JDK9 types, from the day JDK9 was released.
+
+The other, and perhaps more important reason for hiding the Reactive streams interfaces comes back to the first points of this explanation: the fact of Reactive Streams being an SPI, and as such is hard to "get right" in ad-hoc implementations. Thus Akka Streams discourages, the use of the the hard to implement pieces of the underlying infrastructure, and offers simpler, more type-safe, yet equally powerful abstractions for users to work with: GraphStages and operators.
+
```

Comment:
One thing that is missing is that we should mention that we provide 'asPublisher', 'fromSubscriber' that can be used when interoperating with other RS libraries. So that it's misunderstood that we completely hide them and can't be used with other libraries.